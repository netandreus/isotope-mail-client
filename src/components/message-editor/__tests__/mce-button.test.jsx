import React from 'react';
import {shallow} from 'enzyme/build/index';
import MceButton from '../mce-button';

describe('MceButton component test suite', () => {
  test('Snapshot render, should render MceButton', () => {
    const props = {
      className: 'WorkerClass', activeClassName: 'It\'s-a-free-world', iconClassName: 'Upper-class',
      active: true, icon: 'Roy Lichtenstein'};
    const mceButton = shallow(<MceButton {...props} />);
    expect(mceButton).toMatchSnapshot();
  });
  test('Snapshot render with no icon and inactive, should render MceButton', () => {
    const props = {
      className: 'Worker-Class', activeClassName: 'Not-shown', iconClassName: 'Neither', active: false};
    const mceButton = shallow(<MceButton {...props} />);
    expect(mceButton).toMatchSnapshot();
  });
  test('component events, events fired', () => {
    // Given
    const onToggle = jest.fn();
    const props = {
      className: 'WorkerClass', activeClassName: 'It\'s-a-free-world', iconClassName: 'Upper-class', active: true};
    const mceButton = shallow(<MceButton {...props} onToggle={onToggle} />);

    // When
    mceButton.find('button').simulate('mouseDown', {preventDefault: () => {}});
    mceButton.find('button').simulate('mouseUp');

    // Then
    expect(onToggle).toHaveBeenCalledTimes(1);
  });
});
