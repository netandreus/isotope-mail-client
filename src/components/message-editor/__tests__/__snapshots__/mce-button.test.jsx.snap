// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MceButton component test suite Snapshot render with no icon and inactive, should render MceButton 1`] = `
<button
  className="mdc-button Worker-Class "
  onMouseDown={[Function]}
/>
`;

exports[`MceButton component test suite Snapshot render, should render MceButton 1`] = `
<button
  className="mdc-button WorkerClass It's-a-free-world"
  onMouseDown={[Function]}
>
  <i
    aria-hidden="true"
    className="material-icons mdc-button__icon Upper-class"
  >
    Roy Lichtenstein
  </i>
</button>
`;
